From 708a1985265939bd24fd4f821b51c0992691a0ed Mon Sep 17 00:00:00 2001
From: Lin Johnson <johnson.lin@intel.com>
Date: Tue, 4 Sep 2018 16:44:58 +0800
Subject: [PATCH 5/5] Add definition for gem_acess_usrdata

access userdata is kernel feature for IPC purpose
this is necesarry when usage like video deinterlace

Jira: None
Tests: Android boot and deinterlace video playback
---
 include/drm/i915_drm.h | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/include/drm/i915_drm.h b/include/drm/i915_drm.h
index 16e452aa..d8e2b796 100644
--- a/include/drm/i915_drm.h
+++ b/include/drm/i915_drm.h
@@ -319,6 +319,7 @@ typedef struct _drm_i915_sarea {
 #define DRM_I915_PERF_ADD_CONFIG	0x37
 #define DRM_I915_PERF_REMOVE_CONFIG	0x38
 #define DRM_I915_QUERY			0x39
+#define DRM_I915_GEM_ACCESS_USERDATA   0x3c
 
 #define DRM_IOCTL_I915_INIT		DRM_IOW( DRM_COMMAND_BASE + DRM_I915_INIT, drm_i915_init_t)
 #define DRM_IOCTL_I915_FLUSH		DRM_IO ( DRM_COMMAND_BASE + DRM_I915_FLUSH)
@@ -377,6 +378,7 @@ typedef struct _drm_i915_sarea {
 #define DRM_IOCTL_I915_PERF_ADD_CONFIG	DRM_IOW(DRM_COMMAND_BASE + DRM_I915_PERF_ADD_CONFIG, struct drm_i915_perf_oa_config)
 #define DRM_IOCTL_I915_PERF_REMOVE_CONFIG	DRM_IOW(DRM_COMMAND_BASE + DRM_I915_PERF_REMOVE_CONFIG, __u64)
 #define DRM_IOCTL_I915_QUERY			DRM_IOWR(DRM_COMMAND_BASE + DRM_I915_QUERY, struct drm_i915_query)
+#define DRM_IOCTL_I915_GEM_ACCESS_USERDATA     DRM_IOWR(DRM_COMMAND_BASE + DRM_I915_GEM_ACCESS_USERDATA, struct drm_i915_gem_access_userdata)
 
 /* Allow drivers to submit batchbuffers directly to hardware, relying
  * on the security mechanisms provided by hardware.
@@ -1236,6 +1238,21 @@ struct drm_i915_gem_get_tiling {
 	__u32 phys_swizzle_mode;
 };
 
+struct drm_i915_gem_access_userdata {
+       /** Handle of the buffer whose userdata will be accessed */
+       __u32 handle;
+
+       /**
+        * Userdata:  This quantity is user defined
+        */
+       __u32 userdata;
+
+       /**
+        * Write: 0=read userdata, 1=write userdata
+        */
+       __u32 write;
+};
+
 struct drm_i915_gem_get_aperture {
 	/** Total size of the aperture used by i915_gem_execbuffer, in bytes */
 	__u64 aper_size;
-- 
2.17.1

